### PR TITLE
Remove backward-compatibility for many serialization behaviors.

### PR DESCRIPTION
**Pull request by kberg**

### CODE REVIEW

File: src/awards/FundedAward.ts [hunk 1]
```diff
@@ -6,12 +6,9 @@ export interface FundedAward {
   player: Player;
 }
 
-// TODO(kberg): remove references to award and player by 2020-02-01
```
Comment (chosta) [inline src/awards/FundedAward.ts:9]: You did :)

**Review by chosta — APPROVED** [review]

File: tests/Player.spec.ts [hunk 2]
```diff
@@ -232,15 +215,8 @@ describe('Player', function() {
...
-    const newPlayer = Player.deserialize(json as SerializedPlayer, true);
+    const newPlayer = Player.deserialize(json as SerializedPlayer);
 
-    expect(legacyPlayer.color).eq(Color.PURPLE);
     expect(newPlayer.color).eq(Color.PURPLE);
-
-    const legacyPlayerSerialized = legacyPlayer.serialize();
-    const newPlayerSerialized = newPlayer.serialize();
-    expect(legacyPlayerSerialized, 'legacy vs new').to.deep.eq(newPlayerSerialized);
-    expect(legacyPlayerSerialized, 'legacy vs json').to.deep.eq(json);
```
Comment (Lynesth) [inline tests/Player.spec.ts:244]: Please keep a 
```ts
expect(newPlayerSerialized, 'newPlayer vs json').to.deep.eq(json);
```

Comment (kberg) [reply tests/Player.spec.ts:244]: OK. If you mark this as Changes Requested, I can make sure to do it tonight. Or stop someone else from merging this.

**Comment by kberg** [comment]
@bafolts, @Lynesth had asked for modifications.

How much work is involved in restoring this PR in an unmerged state?

**Comment by Lynesth** [comment]
It's just a matter of 2 lines of code in a test, it's more than fine to do it in another PR :)

**Pull request merged**